crates/l2/contracts/src/l1/OnChainProposer.sol

```diff
@@ -137,25 +143,32 @@ contract OnChainProposer is IOnChainProposer, ReentrancyGuard {
     /// @inheritdoc IOnChainProposer
     function commit(
         uint256 blockNumber,
-        bytes32 commitment,
+        bytes32 stateDiffKZGVersionedHash,
         bytes32 withdrawalsLogsMerkleRoot,
-        bytes32 depositLogs
+        bytes32 processedDepositLogs
     ) external override onlySequencer {
         require(
             blockNumber == lastCommittedBlock + 1,
             "OnChainProposer: blockNumber is not the immediate successor of lastCommittedBlock"
         );
+        // TODO: We are using the state diff KZG versioned hash to figure out
+        // whether a block was committed or not. Ideally, we should use the new
+        // state root hash instead.
         require(
-            blockCommitments[blockNumber].commitmentHash == bytes32(0),
+            blockCommitments[blockNumber].stateDiffKZGVersionedHash ==
+                bytes32(0),
             "OnChainProposer: block already committed"
         );
+
         // Check if commitment is equivalent to blob's KZG commitment.
 
-        if (depositLogs != bytes32(0)) {
-            bytes32 savedDepositLogs = ICommonBridge(BRIDGE)
-                .getDepositLogsVersionedHash(uint16(bytes2(depositLogs)));
+        if (processedDepositLogs != bytes32(0)) {
+            bytes32 pendingDepositLogs = ICommonBridge(BRIDGE)
```

Comment:
Here i'd change it from `pendingDepositLogs` to `claimedProcessedDepositLogs`, since what's happening here is we are popping as many deposit logs from the pending queue as the commited block claims it has processed, and making sure that their rolling hash coincides with the one passed to the contract by the sequencer (which then ends up being a public input to the proof)

Reply:
Done in [eef8cb5](https://github.com/lambdaclass/ethrex/pull/2551/commits/eef8cb55c298f53f0736f15c3a6560f966e5c321).